grouper/fe/handlers/permissions_grant_tag.py

```diff
@@ -0,0 +1,73 @@
+from sqlalchemy.exc import IntegrityError
+from grouper.fe.forms import PermissionGrantTagForm
+from grouper.fe.util import GrouperHandler
+from grouper.models.audit_log import AuditLog
+from grouper.permissions import grant_permission_to_tag
+from grouper.models.permission import Permission
+from grouper.constants import TAG_EDIT
+from grouper.models.public_key_tag import PublicKeyTag
+
+
+class PermissionsGrantTag(GrouperHandler):
+    def get(self, name=None):
+        tag = PublicKeyTag.get(self.session, None, name)
+        if not tag:
+            return self.notfound()
+
+        if not self.current_user.has_permission(TAG_EDIT, tag.name):
+            return self.forbidden()
+
+        form = PermissionGrantTagForm()
+        form.permission.choices = [["", "(select one)"]]
+
+        for perm in self.session.query(Permission).all():
+            form.permission.choices.append([perm.name, "{} (*)".format(perm.name)])
+
+        return self.render(
+            "permission-grant-tag.html", form=form, tag=tag,
+        )
+
+    def post(self, name=None):
+        tag = PublicKeyTag.get(self.session, None, name)
+        if not tag:
+            return self.notfound()
+
+        if not self.current_user.has_permission(TAG_EDIT, tag.name):
+            return self.forbidden()
+
+        form = PermissionGrantTagForm(self.request.arguments)
+        form.permission.choices = [["", "(select one)"]]
+
+        for perm in self.session.query(Permission).all():
+            form.permission.choices.append([perm.name, "{} (*)".format(perm.name)])
+
+        if not form.validate():
+            return self.render(
+                "permission-grant-tag.html", form=form, tag=tag,
+                alerts=self.get_form_alerts(form.errors)
+            )
+
+        permission = Permission.get(self.session, form.data["permission"])
+        if not permission:
+            return self.notfound()  # Shouldn't happen.
+
+        try:
+            grant_permission_to_tag(self.session, tag.id, permission.id,
+                                    argument=form.data["argument"])
+        except IntegrityError:
+            form.argument.errors.append(
+                "Permission and Argument already mapped to this tag."
+            )
+            self.session.rollback()
+            return self.render(
+                "permission-grant-tag.html", form=form, tag=tag,
+                alerts=self.get_form_alerts(form.errors),
+            )
+
+        self.session.commit()
```

Comment:
is this commit superfluous? i want to say `grant_pemrission_to_tag` will commit as well.